test/repos/tasks/test_sync_zoo_file.py

```
@@ -0,0 +1,160 @@
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture()
+def generate_repositories(repository_factory):
+    repository_factory(
+        id=1,
+        remote_id=11,
+        name="test_proj1",
+        owner="john_doe1",
+        url="https://github.com/john_doe1/test_proj1",
+        provider="github",
+    )
+
+
+def test_update_or_create_service(generate_repositories):
+    data = {
+        "type": "service",
+        "name": "test_proj1",
+        "owner": "john_doe1",
+        "impact": "profit",
+        "status": "beta",
+        "docs_url": "http://test_proj1/docs",
+        "slack_channel": "#test_proj1",
+        "sentry_project": "http://test_proj1/sentry",
+        "sonarqube_project": "http://test_proj1/sonarqube",
+        "pagerduty_url": "http://test_proj1/pager_duty",
+        "tags": ["tag1", "tag2", "tag3"],
+        "environments": [
+            {
+                "name": "staging",
+                "dashboard_url": "http://staging.test_proj1/dashboard",
+                "service_urls": [
+                    "http://staging.test_proj1/service1",
+                    "http://staging.test_proj1/service2",
+                ],
+                "health_check_url": "http://staging.test_proj1/health_check",
+            },
+            {
+                "name": "production",
+                "dashboard_url": "http://production.test_proj1/dashboard",
+                "service_urls": [
+                    "http://production.test_proj1/service1",
+                    "http://production.test_proj1/service2",
+                ],
+                "health_check_url": "http://production.test_proj1/health_check",
+            },
+        ],
+    }
+
+    proj = {"id": 11, "provider": "github"}
+
+    uut.update_or_create_service(data, proj)
+    try:
+        service = Service.objects.filter(owner=data["owner"], name=data["name"]).first()
+    except Service.DoesNotExist:
+        pytest.fail("Service object not found!!! This should not happen!!!")
```

I wouldn't do this anyway, The error will be propagated, you don't need to catch it and it will be obvious from the error message what caused the error.

---

Yeah I agree on this. I will just let it fail.